File: kirun-js/src/engine/json/schema/validator/exception/SchemaReferenceException.ts
export class SchemaReferenceException extends Error {
    private schemaPath: string;
    private cause?: Error;

    constructor(schemaPath: string, message: string, err?: Error) {
        super(schemaPath.trim() ? schemaPath + '-' + message : message);

        this.schemaPath = schemaPath;
        this.cause = err;
    }

    public getSchemaPath(): string {
        return this.schemaPath;
    }

    public getCause(): Error | undefined {
        return this.cause;
    }
}

File: kirun-py/src/kirun_py/json/schema/validator/exception/schema_reference_exception.py
from __future__ import annotations
from typing import Optional


class SchemaReferenceException(Exception):
    def __init__(self, schema_path: str, message: str = '', cause: Optional[Exception] = None):
        super().__init__(message or f'Schema reference resolution failed at {schema_path}')
        self._schema_path = schema_path
        self._cause = cause

    def get_schema_path(self) -> str:
        return self._schema_path

    def get_cause(self) -> Optional[Exception]:
        return self._cause
